src/spikeinterface/core/generate.py

```diff
@@ -709,14 +714,21 @@ def synthesize_poisson_spike_vector(
     spike_frames[:num_correct_frames] = spike_frames[mask]  # Avoids a malloc
     unit_indices = unit_indices[mask]
 
-    # Sort globaly
-    spike_frames = spike_frames[:num_correct_frames]
-    sort_indices = np.argsort(spike_frames, kind="stable")  # I profiled the different kinds, this is the fastest.
+    train_length = len(unit_indices)
+
+    spike_train = np.zeros(train_length, dtype=minimum_spike_dtype)
+    spike_train["sample_index"] = spike_frames[:num_correct_frames]
+    spike_train["unit_index"] = unit_indices
+
+    spike_train = add_insertions(spike_train, insertions, insertions_replace, rng)
 
-    unit_indices = unit_indices[sort_indices]
-    spike_frames = spike_frames[sort_indices]
+    spike_train.sort(order=["sample_index"])
 
-    return spike_frames, unit_indices
+    segment_length = train_length // segments
+    for i in range(train_length):
+        spike_train["segment_index"][i] = min(i // segment_length, segments - 1)
+
+    return spike_train
```

Comment:
Please don't change the output of a function in an unrelated PR. This might be a good idea but is a big change as we are violating backwards compatbility so it should be discussed in more detail. 

In particular here, I optimized a lot this function to work very well with that and did not intend it to produce structured output. If you want to push for that we should discuss it outside but you can achieve the same result doing that post-processing outside.

Reply:
Sorry @h-mayorquin - I was only thinking about where this function was used in spikeinterface itself. Wasn't thinking about it being exposed to the user. Will change this :)

Reply:
Thanks, you can do all this processing outside of `synthesize_poisson_spike_vector`  where you needed. 

My goal for `synthesize_poisson_spike_vector` is to keep a simple function that can be used in many places. For that purpose, I prefer a simple signature that returns a mostly bare bones object so the function can be easily understood, modified and optimized if necessary. Adding more parameters, options and complexity goes against that purpose and I don't think is really necessary at that level.

Reply:
Hello, should all be reverted now :)